### DISCUSSION TITLE
What extra props do we need to pass to Chakra UI in order to NextJS Links to work?

### DISCUSSION THREAD

**Discussion by SherryH** [chakra-ui/chakra-ui#3821] [Q&A] [answered]
Hi people!
I tried turning Chakra UI elements into links using NextJS Links in my app.
Something I noticed is that:
1. passing `passHref` in the `<Link/>` and wrapping components in `React.forwardRef` is not enough
2. I need to pass additional props using the rest operator for the Link to successfully directs URL.

Here is an example on the [Codesandbox](https://codesandbox.io/s/nextjs-chakra-ui-forked-7pgl9?file=/pages/index.tsx)

If you click on the `About Page` tag, you don't get directed to the About Page.
If you click on the `About Page-Linked` tag, you WILL get directed to the About Page.
The only difference between the 2 components `<MyTag/>` and `<LinkedTag/>` is that extra props are passed into the `<LinkedTag/>` using rest operator.

Can someone shed some light on why this happens?

What are the additional props should we pass to enable the NextJS Link with Chakra UI?

Thanks!!

**Comment by chr-ge** [comment, marked as answer]
You need to give the `href` to the Tag or else it won't work. That's why when you spread the `..rest` of the props it works.

FYI the chakra version in the sandbox is outdated.

**Reply by SherryH** [reply]
Cool @chr-ge Thanks a lot!
